runtime/drivers/duckdb/model_manager.go

```diff
@@ -0,0 +1,176 @@
+package duckdb
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/rilldata/rill/runtime/drivers"
+)
+
+type ModelInputProperties struct {
+	SQL string `mapstructure:"sql"`
+}
+
+func (p *ModelInputProperties) Validate() error {
+	if p.SQL == "" {
+		return fmt.Errorf("missing property 'sql'")
+	}
+	return nil
+}
+
+type ModelOutputProperties struct {
+	Table               string                      `mapstructure:"table"`
+	Materialize         *bool                       `mapstructure:"materialize"`
+	UniqueKey           []string                    `mapstructure:"unique_key"`
+	IncrementalStrategy drivers.IncrementalStrategy `mapstructure:"incremental_strategy"`
+}
+
+func (p *ModelOutputProperties) Validate(opts *drivers.ModelExecutorOptions) error {
+	if opts.Incremental {
+		if p.Materialize != nil && !*p.Materialize {
+			return fmt.Errorf("incremental models must be materialized")
```

Comment:
**Issue**: The error message "incremental models must be materialized" could be more descriptive. It should explain why materialization is required for incremental models. <br> **Fix**: Update the error message to provide more context to the user, explaining why materialization is necessary for incremental models. <br> **Code Suggestion**: 
 ```
 - return fmt.Errorf("incremental models must be materialized")
 + return fmt.Errorf("Incremental models must be materialized to ensure data consistency and support efficient query performance.")
 ```